<a href="https://colab.research.google.com/github/kipsangchepesa/Kipsang-DSP11/blob/main/Moringa_Data_Science_Prep_W3_Independent_Project_2021_08_Kipsang_Nicholas_Data_Analysis_report.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1.1 Importing library

In [ ]:
#Importing pandas library.
import pandas as pd 


#1.2 Loading dataset from excel, csv and Url.

We will perform loading and cleaning of data simultaniously by droping those columns that doesnot matter the most

##### 1.2.1 Loading dataset from excel and csv respectively.

Loading CDR dataset



In [ ]:
#cdr dataset is an excel file therefore pd.read_excel () is useful
CDR_description =pd.read_excel("CDR_description.xlsx")
CDR_description.drop(['Format'], inplace=True, axis=1)
CDR_description.head(4)

Column name                                        Description
0       PRODUCT                                       Voice or SMS
1         VALUE                                      Billing price
2     DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0
3  CELL_ON_SITE  Which cell in the site was used (not needed here)

The dataset are description of columns that are in MTN datasets.Products represents voice or sms and the value is billing price.

In [ ]:
#loading cells geo dataset.
cells_geo = pd.read_csv("cells_geo.csv", sep =";")
#droping columns Unnamed: 0,ZONENAME,	LONGITUDE,	LATITUDE,	REGION,	AREA,	CELL_ID and 	SITE_COD
cells_geo.drop(['Unnamed: 0','ZONENAME','LONGITUDE','LATITUDE','REGION','AREA','CELL_ID','SITE_CODE','DECOUPZONE'],inplace=True, axis=1)
cells_geo.head(500)

VILLES      STATUS LOCALISATION
0     ADJAME  In Service      ABIDJAN
1     ADJAME  In Service      ABIDJAN
2     ADJAME  In Service      ABIDJAN
3    ASSINIE  In Service    INTERIEUR
4    ASSINIE  In Service    INTERIEUR
..       ...         ...          ...
495    ABOBO  In Service      ABIDJAN
496    ABOBO  In Service      ABIDJAN
497    ABOBO  In Service      ABIDJAN
498    ABOBO  In Service      ABIDJAN
499    ABOBO  In Service      ABIDJAN

[500 rows x 3 columns]

In [ ]:
#CONCAT Telcom dataset 1 and cellsgeo dataset to have 
newdataset1= pd.concat([cells_geo,Telcom_dataset1], axis=1)


VILLES      STATUS  ...               DATETIME CELL_ON_SITE
0      ADJAME  In Service  ...  2012-05-06 23:04:37.0            1
1      ADJAME  In Service  ...  2012-05-06 23:05:25.0            0
2      ADJAME  In Service  ...  2012-05-06 23:05:41.0            0
3     ASSINIE  In Service  ...  2012-05-06 23:05:42.0            1
4     ASSINIE  In Service  ...  2012-05-06 23:06:29.0            1
...       ...         ...  ...                    ...          ...
4996      NaN         NaN  ...  2012-05-07 00:01:54.0            0
4997      NaN         NaN  ...  2012-05-07 00:01:54.0            1
4998      NaN         NaN  ...  2012-05-07 00:01:54.0            1
4999      NaN         NaN  ...  2012-05-07 00:01:54.0            0
5000      NaN         NaN  ...  2012-05-07 00:01:54.0            2

[5001 rows x 7 columns]

In [ ]:
#loading the cell geo desrciption dataset contains columns that 
cells_geo_description =pd.read_excel("cells_geo_description.xlsx")
#droping the format column in the dataset.
cells_geo_description.drop(['Format'],inplace=True,axis=1)
cells_geo_description.head(11)

Column name                                        Description
0         VILLES                                               City
1         STATUS                                  In Service or not
2   LOCALISATION                                  in ABIDJAN or not
3     DECOUPZONE                                  Geographical Zone
4       ZONENAME                                       Name of Zone
5      LONGITUDE                                          Longitude
6       LATITUDE                                           Latitude
7         REGION                                             Region
8           AREA                                               Area
9        CELL_ID                                     ID of the cell
10     SITE_CODE  Site (there are several cells per site, severa...

##### 1.2.1 Loading dataset from URL.

We will perform loading and cleaning of data simultaniously by droping those columns that doesnot matter the most. 

In [ ]:
#Loading Telcom dataset 1
Telcom_dataset1 = pd.read_csv("http://bit.ly/Telcom_dataset1")
Telcom_dataset1.drop(["DW_A_NUMBER_INT","DW_B_NUMBER_INT","COUNTRY_A","COUNTRY_B","CELL_ID","SITE_ID"],inplace=True, axis=1)
Telcom_dataset1

PRODUTC  VALUE               DATETIME CELL_ON_SITE
0      Voice      0  2012-05-06 23:04:37.0            1
1        sms      0  2012-05-06 23:05:25.0            0
2        sms     25  2012-05-06 23:05:41.0            0
3      Voice     56  2012-05-06 23:05:42.0            1
4      Voice     14  2012-05-06 23:06:29.0            1
...      ...    ...                    ...          ...
4996     sms    100  2012-05-07 00:01:54.0            0
4997   Voice      1  2012-05-07 00:01:54.0            1
4998   Voice      0  2012-05-07 00:01:54.0            1
4999   Voice      0  2012-05-07 00:01:54.0            0
5000   Voice      7  2012-05-07 00:01:54.0            2

[5001 rows x 4 columns]

In [ ]:
#Loading Telcom dataset 2
Telcom_dataset2 = pd.read_csv("http://bit.ly/Telcom_dataset2")
Telcom_dataset2.drop(["DW_A_NUMBER","DW_B_NUMBER","COUNTRY_A","COUNTRY_B","CELL_ID","SITE_ID"],inplace=True, axis=1)
Telcom_dataset2

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE
0        sms      0  2012-05-07 23:02:06.0            2
1      Voice      0  2012-05-07 23:03:44.0            1
2        sms      0  2012-05-07 23:04:06.0            0
3      Voice     37  2012-05-07 23:04:48.0            1
4        sms     25  2012-05-07 23:04:58.0            1
...      ...    ...                    ...          ...
4996     sms      0  2012-05-08 00:01:53.0            8
4997     sms     25  2012-05-08 00:01:53.0            0
4998     sms      0  2012-05-08 00:01:53.0            0
4999    data      0  2012-05-08 00:01:53.0            _
5000   Voice      0  2012-05-08 00:01:53.0            0

[5001 rows x 4 columns]

In [ ]:
#Loading Telcom dataset 3
Telcom_dataset3 = pd.read_csv("http://bit.ly/Telcom_dataset3")
Telcom_dataset3.drop(["DW_A_NUMBER_INT","DW_B_NUMBER_INT","COUNTRY_A","COUNTRY_B","CELLID","SIET_ID"],inplace=True, axis=1)
Telcom_dataset3

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE
0      Voice     61  2012-05-08 23:01:28.0            2
1        sms      0  2012-05-08 23:01:31.0            1
2      Voice     10  2012-05-08 23:02:59.0            1
3        sms      0  2012-05-08 23:03:04.0            0
4      Voice     22  2012-05-08 23:03:16.0            4
...      ...    ...                    ...          ...
4996   Voice      0  2012-05-09 00:01:43.0            3
4997   Voice     62  2012-05-09 00:01:43.0            1
4998   Voice     32  2012-05-09 00:01:43.0            1
4999   Voice     30  2012-05-09 00:01:43.0            2
5000   Voice     60  2012-05-09 00:01:43.0            2

[5001 rows x 4 columns]

# 1.3 The the most used city for the three days.

In [ ]:
#concate cell geo dataset with Telcom dataset1 to have
newdataset1=pd.concat([cells_geo,Telcom_dataset1], axis=1)
newdataset1['VILLES'].value_counts()

COCODY         354
YOPOUGON       295
ABOBO          255
MARCORY        141
TREICHVILLE    132
              ... 
KEIBLA           3
MEMNI            2
KOTOULA          2
OUANGOLO         1
DANANON          1
Name: VILLES, Length: 504, dtype: int64

In [ ]:
#concate cell geo dataset with Telcom dataset2 to have
newdataset2=pd.concat([cells_geo,Telcom_dataset2], axis=1)
newdataset2['VILLES'].value_counts()

COCODY         354
YOPOUGON       295
ABOBO          255
MARCORY        141
TREICHVILLE    132
              ... 
KEIBLA           3
MEMNI            2
KOTOULA          2
OUANGOLO         1
DANANON          1
Name: VILLES, Length: 504, dtype: int64

In [ ]:
#concate cell geo dataset with Telcom dataset3 to have
newdataset3=pd.concat([cells_geo,Telcom_dataset3], axis=1)
newdataset3['VILLES'].value_counts()

COCODY         354
YOPOUGON       295
ABOBO          255
MARCORY        141
TREICHVILLE    132
              ... 
KEIBLA           3
MEMNI            2
KOTOULA          2
OUANGOLO         1
DANANON          1
Name: VILLES, Length: 504, dtype: int64

As seen in  the dataset the answer as to which city was most used in the 3 days is COCODY which appear to have been used the most by 354 times in collecting the MTN CDR dataset for the three days.

# 1.4 Cities were the most used during business and home hours.

In [ ]:
#Grouping by product to check the most city with high billing price in each product.
newdataset1.groupby(["PRODUTC","VILLES","STATUS","DATETIME"], sort= True).sum().head(1000)

VALUE
PRODUTC VILLES           STATUS     DATETIME                    
Voice   ABENGOUROU       In Service 2012-05-06 23:15:34.0     14
                                    2012-05-06 23:15:41.0    152
                                    2012-05-06 23:15:44.0    138
                                    2012-05-06 23:15:46.0     30
                                    2012-05-06 23:16:02.0    192
...                                                          ...
sms     ABONGOUA         In Service 2012-05-06 23:30:14.0      0
        ABOUDEKOUASSIKRO In Service 2012-05-06 23:30:17.0      0
                                    2012-05-06 23:30:24.0      0
        ABRONAMOUE       In Service 2012-05-06 23:31:34.0      0
        ADAOU            In Service 2012-05-06 23:33:07.0      0

[1000 rows x 1 columns]

In [ ]:
#Getting the mean of the three products.
newdataset1.groupby(["PRODUTC"]).mean()

VALUE
PRODUTC           
Voice    49.062190
data      8.713755
sms       6.821772

In [ ]:
newdataset2.groupby(["PRODUCT","VILLES","STATUS","DATE_TIME"], sort= True).sum().head(1000)

VALUE
PRODUCT VILLES     STATUS     DATE_TIME                   
Voice   ABENGOUROU In Service 2012-05-07 23:13:29.0    138
                              2012-05-07 23:13:34.0     54
                              2012-05-07 23:13:35.0      0
                              2012-05-07 23:13:37.0     38
                              2012-05-07 23:13:38.0      0
...                                                    ...
sms     ABOBO      In Service 2012-05-07 23:28:19.0      0
                              2012-05-07 23:37:39.0      0
                              2012-05-07 23:37:48.0     50
                              2012-05-07 23:37:50.0     50
                              2012-05-07 23:39:13.0      0

[1000 rows x 1 columns]

In [ ]:
newdataset2.groupby(["PRODUCT"]).mean()

VALUE
PRODUCT           
Voice    54.294858
data      7.180157
sms       5.458566

In [ ]:
newdataset3.groupby(["PRODUCT","VILLES","STATUS","DATE_TIME"], sort= True).sum().head(1000)

VALUE
PRODUCT VILLES     STATUS     DATE_TIME                   
Voice   ABENGOUROU In Service 2012-05-08 23:14:03.0     88
                              2012-05-08 23:14:26.0    159
                              2012-05-08 23:14:42.0      0
                              2012-05-08 23:14:44.0     56
                              2012-05-08 23:14:57.0      0
...                                                    ...
sms     BONOUFLA   In Service 2012-05-09 00:00:06.0      0
        BOUADIKRO  In Service 2012-05-09 00:00:07.0      2
        BOUAFLE    In Service 2012-05-09 00:00:07.0    100
        BOUAKE     In Service 2012-05-08 23:46:41.0      0
                              2012-05-08 23:46:47.0      0

[1000 rows x 1 columns]

In [ ]:
newdataset3.groupby(["PRODUCT"]).mean()

VALUE
PRODUCT           
Voice    65.047372
data      6.861472
sms       9.844958

The most used cities used during home hours are ABENGOUROU, ABOBO and BLESSEGUE in the three days but no city were used during business hour.